client/app/validators/required-if-selected.js

```diff
@@ -3,12 +3,12 @@ import {
 } from 'ember-changeset-validations/validators';
 
 export default function validatePresenceIf(options) {
-  const { withValue, on } = options;
+  const { withValue, on, comparison } = options;
 
   return (...args) => {
     const [,,, changes, content] = args;
     const target = Object.prototype.hasOwnProperty.call(changes, on) ? changes[on] : content[on];
-    const hasMatchingWith = target === withValue;
+    const hasMatchingWith = comparison === 'not' ? target !== withValue : target === withValue;
```

Comment:
I think if we want to have this behavior we can be a little more creative with the API. What i've seen with some APIs is this:

by default, accept a primitive:
```javascript
{ withValue: 123 }
```

or, accept a function, that's provided some context:

```javascript
{
  withValue: (target) => {
    return target === null || target === undefined;
  }
}
```

In your code here, you would do something like this:

```javascript
let hasMatchingWith;

if (typeof withValue === 'function') {
  hasMatchingWith = withValue(target); // now call it with the derived context
} else {
  hasMatchingWith = target === withValue;
}
```

untested code but this could work.


then get rid of your "comparison" from the API space since this will give you the flexibility you need.